project_model_to_task/models/project.py

```diff
@@ -22,7 +22,7 @@ def _authorised_models(self):
         """ Inherit this method to add more models depending of your
             modules dependencies
         """
-        models = self.env['ir.model'].search([('model', '!=', 'project.task')])
+        models = self.env['ir.model'].search([])
```

Comment:
I removed the original domain to allow the use of project.task as any other model. Thus we can link a task to an other task.